### PR TITLE
Improve logger, add grad norm + learning rate

### PR DESCRIPTION
**Pull request by gordicaleksa**
Other than loss, gradient norm and learning rate tell a lot of information about the training dynamics:
1) modified the logger to support logging these 2 quantities
2) updated jupyter notebook parser

### CODE REVIEW

**Comment by karpathy** [comment]
agree!
